docs/PLUGIN_LIFECYCLE.md

```diff
@@ -0,0 +1,86 @@
+<!--
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+# Plugin life cycle
+
+A snap plugin can be in a `Loaded` or `Running` state.  A plugin can be loaded
+in the following two ways.
+
+1. `snapd` was started with an auto discover path `snapd -a /etc/snapd/plugins`
+2. A user loads a plugin through the REST API or using the snapctl 
+    * `curl -F file=@snap-plugin-publisher-file http://localhost:9191/v1/plugins`
+    * `snapctl plugin load snap-plugin-publisher-file` 
+
+A plugin transitions to a `running` state when a task is started that uses the 
+plugin.  This is also called a plugin subscription.  
+
+## What happens when a plugin is loaded
+
+When a plugin is loaded snapd takes the following steps.
+
+1. Handshakes with the plugin by reading it's stdout
+2. Updates the metric catalog by calling the plugin over RPC
+    * `GetMetricTypes` returns the metrics the plugin provides
+    * `GetConfigPolicy` returns the conf policy that the plugin needs
+3. The plugin is stopped
+
+It should be emphasized that when a plugin is loaded it is started but stopped 
+as soon as the metric catalog has been updated.  
+
+## What happens when a task is started
+
+When a task is started the plugins that it references are started and subscribed 
+to.   The following steps are taken when a task is created and started.
+
+1. On **task creation** the task is validated (`snapctl task create -t mytask.yml
+--no-start`) 
+    * The schedule is validated
+    * The config provided for the metrics (collectors), processors and 
+    publishers are validated
+2. On **task starting** the plugins are started (`snapctl task start <TASK_ID>`)
+3. Subscriptions for each plugin referenced by the task are incremented 
+
+## Diving deeper
+
+**Task started** - When a task is started the plugins which are referenced by 
+the workflow are subscribed to and a 'subscription group' is created.  A 
```

Comment:
first use of the term "workflow" might be confusing here.  I'd try to be more specific if possible.  Maybe the "task manifest" or "workflow manifest" or are you referring to "workflow" in a more general case?